app/src/marketplace/containers/EditProductPage/BeneficiaryAddress.jsx

```diff
@@ -18,22 +28,85 @@ type Props = {
     className?: string,
 }
 
+const EMPTY = []
+
 const BeneficiaryAddress = ({ address, onChange, disabled, className }: Props) => {
     const { isValid, message } = useValidation('beneficiaryAddress')
     const { isTouched } = useContext(ValidationContext)
     const priceTouched = isTouched('pricePerSecond') || isTouched('beneficiaryAddress')
     const invalid = priceTouched && !isValid
 
+    const { copy } = useCopy()
+
+    const dispatch = useDispatch()
+
+    const integrationKeys = useSelector(selectEthereumIdentities) || EMPTY
+
+    const integrationKeysFiltered = useMemo(() => (
+        integrationKeys.filter(({ json }) => json && json.address)
+    ), [integrationKeys])
+
+    const onCopy = useCallback((value: string) => {
+        copy(value)
+
+        Notification.push({
+            title: 'Copied',
+            icon: NotificationIcon.CHECKMARK,
+        })
+    }, [copy])
+
+    useEffect(() => {
+        dispatch(fetchIntegrationKeys())
+    }, [dispatch])
+
+    const accountAddress = useAccountAddress()
+
+    const useCurrentWalletAddress = useCallback(() => {
+        if (accountAddress) {
+            onChange(accountAddress)
+        }
+    }, [accountAddress, onChange])
+
     return (
-        <form autoComplete="off">
+        <form autoComplete="off" onSubmit={(e) => e.preventDefault()}>
             <label
                 htmlFor="beneficiaryAddress"
                 className={cx(styles.root, styles.BeneficiaryAddress, className)}
             >
                 <strong>
                     <Translate value="editProductPage.setPrice.setRecipientEthAddress" />
                 </strong>
-                <div>
+                <ActionsDropdown
+                    actions={[
+                        <DropdownActions.Item
+                            key="useCurrent"
+                            onClick={useCurrentWalletAddress}
+                            disabled={!accountAddress}
+                        >
+                            Use current wallet address
+                        </DropdownActions.Item>,
+                        ...integrationKeysFiltered.map(({ id, name, json }) => (
+                            <DropdownActions.Item
+                                key={id}
+                                onClick={() => onChange(json.address)}
+                            >
+                                {`Use ${truncate(json.address, {
+                                    maxLength: 15,
```

Comment:
Truncate could also be used for the display value since we have the `copy` function in the dropdown.